client/src/featureform/train_test_split.py

```diff
@@ -0,0 +1,212 @@
+            get_numpy_array_type,
+            proto_type_to_np_type,
+        )
+
+        types = [
+            (feature.WhichOneof("value"), proto_type_to_np_type(feature))
+            for feature in row.features
+        ]
+        numpy_array_type = get_numpy_array_type(list(map(lambda x: x[1], types)))
+        label_proto_type = row.label.WhichOneof("value")
+        return _NpProtoTypeParser(numpy_array_type, types, label_proto_type)
+
+    def parse_features(self, rows: List[serving_pb2.Value]) -> List[Any]:
+        return [
+            getattr(feature, self.feature_proto_types[i][0])
+            for i, feature in enumerate(rows)
+        ]
+
+    def parse_label(self, row: serving_pb2.Value) -> Any:
+        return getattr(row, self.label_proto_type)
+
+
+class TrainTestSplitIterator:
+    def __init__(
+        self,
+        split_stream: _SplitStream,
+        batch_size: int,
+        request_type: serving_pb2.RequestType,
+    ):
+        self.request_type = request_type
+        self._split_stream = split_stream
+        self._batch_size = batch_size
+        self._complete = False
+        self._np_type_parser = None
+
+    @staticmethod
+    def train_iter(split_stream, batch_size):
+        return TrainTestSplitIterator(
+            split_stream, batch_size, serving_pb2.RequestType.TRAINING
+        )
+
+    @staticmethod
+    def test_iter(split_stream, batch_size):
+        return TrainTestSplitIterator(
+            split_stream, batch_size, serving_pb2.RequestType.TEST
+        )
+
+    def __iter__(self):
+        return self
+
+    def __next__(self) -> Tuple[np.ndarray, np.ndarray]:
+        if self._complete:
+            raise StopIteration
+
+        batch_features_list = []
+        batch_labels_list = []
+
+        result = self._split_stream.send_request(self.request_type)
+        data = result.data
+
+        if self._np_type_parser is None and data.rows:
+            self._np_type_parser = _NpProtoTypeParser.init_types(data.rows[0])
+
+        for i, row in enumerate(data.rows):
+            batch_features_list.append(
+                self._np_type_parser.parse_features(row.features)
+            )
+            batch_labels_list.append(self._np_type_parser.parse_label(row.label))
+
+        if result.iterator_done:
+            self._complete = True
+
+        if not batch_features_list:  # If no data was collected
+            raise StopIteration
+
+        batch_features = np.array(
+            batch_features_list, dtype=self._np_type_parser.nptype
+        )
+        batch_labels = np.array(batch_labels_list)
+
+        return batch_features, batch_labels
+
+
+class TrainingSetTestSplit:
+    """
+    Returns two iterators, one for the training set and one for the test set, in that order
+    """
+
+    def __init__(
+        self,
+        stub,
+        name: str,
+        version: str,
+        test_size: float,
+        train_size: float,  # unused for now
```

Comment:
if unused remove from `init`?

Reply:
sterling said to keep it and I don't care enough lol